frontend/src/components/Dashboard/DataIngestion/DataIngestion.tsx

```diff
@@ -35,26 +37,27 @@ export const DataIngestion = ({ modelId }: DataIngestionProps) => {
 
   const { graphData, isLoading } = useDataIngestion(modelId, selectedPointType, timeValue);
 
-  const labelsArr = ['Samples', 'Labels', 'Missing Labels'];
+  const handleTime = (value: unknown) => {
+    if (typeof value !== 'string' && typeof value !== 'number') return;
 
-  useEffect(() => {
-    if (+timeValue <= 3600) {
+    const newTimeValue = +value;
+    const newTimeIndex = timeOptions.findIndex(time => time.value === newTimeValue);
+
+    if (newTimeValue <= frequencyValues.HOUR) {
       setMinTimeUnit('minute');
-    } else if (+timeValue <= frequencyValues.DAY) {
+    } else if (newTimeValue <= frequencyValues.DAY) {
       setMinTimeUnit('hour');
-    } else {
+    } else if (newTimeValue <= frequencyValues.WEEK) {
       setMinTimeUnit('day');
+    } else if (newTimeValue <= frequencyValues.MONTH) {
+      setMinTimeUnit('week');
+    } else {
+      setMinTimeUnit('month');
     }
-  }, [timeValue]);
```

Comment:
why no useEffect? I remember without it it caused bugs

Reply:
It caused a UI crash and we don't need it.